src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreAndGenerateDepsjsonDotNetCliToolReference.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Restore.Tests
+{
+    public class GivenThatWeWantToRestoreAndGenerateDepsjsonDotNetCliToolReference : SdkTest
+    {
+        public GivenThatWeWantToRestoreAndGenerateDepsjsonDotNetCliToolReference(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [Fact]
+        public void It_can_generate_correct_deps_json()
+        {
+            TestProject toolProject = new TestProject()
+            {
+                Name = "DotNetCliToolReferenceProject",
+                IsSdkProject = true,
+                IsExe = true,
+                TargetFrameworks = "netcoreapp1.0",
+            };
+
+            toolProject.AdditionalProperties.Add("PackageTargetFallback", "$(PackageTargetFallback);portable-net45+win8+wp8+wpa81;");
```

Comment:
What's this necessary for?  A comment would be good.

Reply:
it is necessary. it is in the aspnet template. I added comment